### PR TITLE
rewrite the Components section landing page to explain better how to write a component

### PR DESCRIPTION
**Pull request by doudou**

### CODE REVIEW

File: source/components/index.html.md [hunk 1]
```diff
@@ -1,17 +1,161 @@
...
+structure that is specific to the component being developed.
+{: .note}
+
+**Guidelines**
+
+- **group together data that belong together**. Do not create 5 ports to get a piece of
+  data because, for instance, you could not find an existing data structure for that
+  combination. Re-combining data that belong together is difficult in an asynchronous
+  system like Rock.
+- **port names need to be unique**, an input cannot be named the same as an output
```
Comment (caioaamaral) [inline source/components/index.html.md:124]: so maybe we could standardize using `_in` and `_out` suffix to avoid naming collision

**Review by caioaamaral — APPROVED** [review]
Very well-written documentation!

File: source/components/index.html.md (hunk 1 above)
Comment (doudou) [reply source/components/index.html.md:124]: I personally hate the idea, and in practice it's very rarely an issue.

File: source/components/index.html.md [hunk 2]
```diff
@@ -1,17 +1,161 @@
...
 
+Within Rock, components are _implemented_ in C++. They are also _specified_ in a
+Ruby domain-specific language that is processed by a code generation tool,
+**oroGen**. This tool ensures that the component's interface matches its
+specification. It also removes most of the crude boilerplate-writing code that
+is the declaration in C++ of the component interfaces.
+
+From a package point of view, components are defined in an orogen package. The
+orogen packages are all placed in the `/orogen/` subdirectory of one of the
+[package categories](../workspace/conventions.html). This page will not dwell
```
[the remark this answers is not in the document]
Comment (doudou) [reply source/components/index.html.md:23]: You mean, an example ?

File: source/components/importing_types.html.md [hunk 3]
```diff
@@ -42,11 +42,18 @@ import_types_from "hokuyo/Statistics.hpp"
 (i.e. `hokuyo` for `drivers/hokuyo`).
 {: .note}
 
+<div>
 **Important** The `using_library "library_name"` stanza
 implicitly create a dependency between the oroGen package you're working on
 and the library package. This dependency **must** be made explicit by adding
 the corresponding `<depend name="..." />` line to the oroGen package's
```
[the remark this answers is not in the document]
Comment (doudou) [reply source/components/importing_types.html.md:49]: Not here, no. It's really something that only happens in Syskit.

File: source/components/index.html.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (doudou) [reply source/components/index.html.md:23]: Ping @caioaamaral

File: source/components/index.html.md [hunk 4]
```diff
@@ -1,17 +1,161 @@
...
 {:.no_toc}
 
 - TOC
 {:toc}
 
+Within Rock, components are _implemented_ in C++. They are also _specified_ in a
+Ruby domain-specific language that is processed by a code generation tool,
+**oroGen**. This tool ensures that the component's interface matches its
+specification. It also removes most of the crude boilerplate-writing code that
+is the declaration in C++ of the component interfaces.
```
[the remark this answers is not in the document]
Comment (doudou) [reply source/components/index.html.md:19]: Done

File: source/components/index.html.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (caioaamaral) [reply source/components/index.html.md:23]: > You mean, an example ?

yes, just a mock example. I was imagining a a `tree` like output

[the remark this answers is not in the document]
Comment (doudou) [reply source/components/index.html.md]: Just did. Can you check it out ?

[the remark this answers is not in the document]
Comment (caioaamaral) [reply source/components/index.html.md]: looks good to me

**Comment by doudou** [comment]
OK. I'm merging and publishing this.

**Pull request merged**